v2/devdoc/log_context_requirements.md

```diff
@@ -0,0 +1,152 @@
+# `log_context` requirements
+
+`log_context` encapsulates storing a series of properties and their values (that meant to be used when logging is done).
+
+A `log_context` holds a count of properties and the backing memory to hold the properties name and values.
+
+## Exposed API
+
+```
+typedef struct LOG_CONTEXT_TAG* LOG_CONTEXT_HANDLE;
+
+uint32_t log_context_get_property_value_pair_count(LOG_CONTEXT_HANDLE log_context);
+const LOG_CONTEXT_PROPERTY_VALUE_PAIR* log_context_get_property_value_pairs(LOG_CONTEXT_HANDLE log_context);
+
+#define LOG_CONTEXT_CREATE(dest_log_context, parent_context, ...) \
+    ...
+
+#define LOG_CONTEXT_DESTROY(log_context) \
+    ...
+
+#define LOG_MAX_STACK_DATA_SIZE                 1024
+#define LOG_MAX_STACK_PROPERTY_VALUE_PAIR_COUNT 64
+```
+
+## LOG_CONTEXT_CREATE
+
+```c
+#define LOG_CONTEXT_CREATE(dest_log_context, parent_context, ...) \
+    ...
+```
+
+`LOG_CONTEXT_CREATE` creates a dynamically allocated log context.
+
+`...` is a list of `LOG_CONTEXT_PROPERTY`, `LOG_CONTEXT_STRING_PROPERTY` or `LOG_CONTEXT_NAME` entries.
```

Comment:
>LOG_CONTEXT_NAME [](http://example.com/codeflow?start=77&length=16)

can you have multiple of LOG_CONTEXT_NAMEs?

Reply:
Nope.